aries_cloudagent/protocols/present_proof/indy/pres_exch_handler.py

```diff
@@ -120,14 +120,17 @@ async def _get_ledger_objects(self, credentials: dict):
             if credential.get("rev_reg_id"):
                 revocation_registry_id = credential["rev_reg_id"]
                 if revocation_registry_id not in revocation_registries:
+                    rev_reg = (
+                        await anoncreds_registry.get_revocation_registry_definition(
+                            self._profile, revocation_registry_id
+                        )
+                    ).revocation_registry.serialize()
+                    # add id to the serialized rev_reg
+                    rev_reg["id"] = revocation_registry_id
                     revocation_registries[
                         revocation_registry_id
                     ] = RevocationRegistry.from_definition(
-                        (
-                            await anoncreds_registry.get_revocation_registry_definition(
-                                self._profile, revocation_registry_id
-                            )
-                        ).revocation_registry.serialize(),
+                        rev_reg,
                         True,
                     )
         return schemas, cred_defs, revocation_registries
```

Comment:
We should replace use of `RevocationRegistry` with `RevRegDef`; this causes a small cascade of changes needed up the stack from where this method is called but it shouldn't be too painful of a refactor. See https://github.com/Indicio-tech/aries-cloudagent-python/commit/81ab818cd4a2d5be142c4f4abb1bbf62386e5e14#diff-fd952db170fd7d5742268e460f9e739c3e9106a06f4e6e2556a634074841977aL447-R396

Reply:
We will address this in a separate set of changes